src/io/iohk/scalanet/peergroup/PeerGroup.scala

```diff
@@ -3,15 +3,25 @@ package io.iohk.scalanet.peergroup
 import java.nio.ByteBuffer
 
 import cats.data.Kleisli
+import io.iohk.decco.Codec
 import io.iohk.scalanet.messagestream.MessageStream
+import io.iohk.scalanet.peergroup.ControlEvent.InitializationError
 import monix.eval.Task
 
 import scala.language.higherKinds
 
 sealed trait PeerGroup[A, F[_]] {
+  def initialize(): F[Unit] = ???
   def sendMessage(address: A, message: ByteBuffer): F[Unit]
   def shutdown(): F[Unit]
   def messageStream(): MessageStream[ByteBuffer]
+  val processAddress: A
+  val decoderTable: DecoderTable = new DecoderTable()
```

Comment:
Is this supposed to be public and needs a concrete implementation? I guess that it can be private.

Reply:
Thanks. Made it package private.